services/autoscaling/src/simcore_service_autoscaling/background_task.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+import contextlib
+import datetime
+import logging
+from typing import Awaitable, Callable
+
+from servicelib.logging_utils import log_context
+
+logger = logging.getLogger(__name__)
+
+
+async def _repeated_scheduled_task(
+    task: Callable[..., Awaitable[None]],
+    *,
+    interval: datetime.timedelta,
+    task_name: str,
+    **kwargs,
+):
+    while await asyncio.sleep(interval.total_seconds(), result=True):
+        try:
+            with log_context(logger, logging.DEBUG, msg=f"Run {task_name}"):
+                await task(**kwargs)
+        except asyncio.CancelledError:
+            logger.info("%s cancelled", task_name)
+            raise
+        except Exception:  # pylint: disable=broad-except
+            logger.exception("Unexpected error in %s, restarting...", task_name)
+
+
+async def start_background_task(
```

Comment:
At some point we could put this in ``servicelib.fastapi`` 
Q? how does this differ from fastapi's background task? In which chases should we prefer the latter vs an asyncio.task? NOTE: just if you have the answer on the top of your head or just send me to RTFM! ;-)

Reply:
the fastapi background task is not running while the app runs but after a request. here I need is a background task while the app is up until it shuts down